### DISCUSSION TITLE
Integration testing

### DISCUSSION THREAD

**Discussion by rossburton** [openssl/openssl#24932] [Q&A]
I mentioned this in passing in https://github.com/openssl/openssl/discussions/24925 but thought it deserved to be a new discussion.

Is there a recommended way to perform integration testing of OpenSSL _on the target_, where you don't have the openssl build tree?  Right now we're copying large chunks of the build tree to the target and running the unit tests, but that needs to be manually maintained as there's no "install the test suite" command.

**Comment by nhorman** [comment]
honestly, an `install test_sw` target might be a reasonable solution, but I don't think thats going to help you that much.  The executable files alone are 1.1GB in the local build i have here.  Any other ancillary data to build/run the tests (extra perl scripts, data info, etc), appear to relatively take up about 38M.  

Is there a target size you are trying to hit?  I ask because just tarring up the test directory reduces overall size from 1.1G to 432M

**Reply by rossburton** [reply]
The size is because every test executable statically links to the libraries, in https://github.com/openssl/openssl/discussions/24925 I muse on the idea of uninstalled libraries without visibility controls, purely for the test suite to dynamically link to. That would make link faster and use less disk space.

My question here is simply is there a good existing integration test suite that isn't the test/ tree inside openssl itself?

**Reply by nhorman** [reply]
Hmm, I'm afraid I don't think there is.  Best thing I could point you to would be the GnuTLS and NSS interop test job in .github/workflows/interop-tests.yml .  It does high level ssl interop testing for SSL between OpenSSL/GnuTLS and NSS, but it uses several container builds you may need to tease apart
